components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/DeviceManagementServiceProviderImpl.java

```diff
@@ -571,4 +571,14 @@ public Operation getOperation(int operationId) throws OperationManagementExcepti
         }
         return devicesOfRole;
     }
+
+    @Override
+    public int getDeviceCount() throws DeviceManagementException {
+        try {
+            int deviceCount = this.deviceDAO.getDeviceCount();
+            return deviceCount;
+        } catch (DeviceManagementDAOException e) {
+            throw new DeviceManagementException("Error occurred while obtaining devices all devices", e);
```

Comment:
Fixed in commit [1]
[1] https://github.com/wso2/carbon-device-mgt/commit/60960d80b4cfa1e7bf16c3fffb10f9295e2185ae